File: 2025/Company/Company.Web/Modules/ServerTypes/Company/AddressColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AddressRow } from "./AddressRow";

export interface AddressColumns {
    AddressId: Column<AddressRow>;
    AddressLine1: Column<AddressRow>;
    AddressLine2: Column<AddressRow>;
    City: Column<AddressRow>;
    CountryId: Column<AddressRow>;
    ProvinceId: Column<AddressRow>;
    PostalCode: Column<AddressRow>;
}

export class AddressColumns extends ColumnsBase<AddressRow> {
    static readonly columnsKey = 'Company.Address';
    static readonly Fields = fieldsProxy<AddressColumns>();
}

File: 2025/Company/Company.Web/Modules/Company/Address/AddressColumns.cs
namespace Company.Company.Columns;

[ColumnsScript("Company.Address")]
[BasedOnRow(typeof(AddressRow), CheckNames = true)]
public class AddressColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int AddressId { get; set; }
    [EditLink, Width(200)]
    public string AddressLine1 { get; set; }
    [Width(200)]
    public string AddressLine2 { get; set; }
    [Width(100)]
    public string City { get; set; }
    [Width(100)]
    //public string CountryId { get; set; }
    public string CountryName {  get; set; }
    [Width(100)]
    //public int ProvinceId { get; set; }
    public int ProvinceName { get; set; }
    [Width(100)]
    public string PostalCode { get; set; }
}